js/app.js

```diff
@@ -1,12 +1,56 @@
 $(function(){
   var doc = $(document);
 
+  // Configuration for smooth scrolling
+  var scrollDuration = 500;
+  var headerOffset = 120;
+
+  // Smooth scroll function
+  function smoothScrollToElement(target) {
+    var $target = $(target);
+    
+    // Handle IDs with dots (like express.urlencoded)
+    if (!$target.length && typeof target === 'string' && target.indexOf('.') > -1) {
+      var targetId = target.replace('#', '');
+      $target = $('[id="' + targetId + '"]');
+    }
+    
+    if ($target.length) {
+      $('html, body').animate({
+        scrollTop: $target.offset().top - headerOffset
+      }, scrollDuration);
+      return true;
+    }
+    return false;
+  }
+
   // top link
   $('#top').click(function(e){
     $('html, body').animate({scrollTop : 0}, 500);
     return false;
   });
 
+  // Smooth scrolling for all anchor links
+  $(document).on('click', 'a[href^="#"]:not([href="#"]):not([href="#top"])', function(e) {
```

Comment:
```suggestion
  $("#api-doc").on('click', 'a[href^="#"]:not([href="#"]):not([href="#top"])', function(e) {
```
We should minimize the scope of this click event. Currently, $(document).on('click'...) attaches the handler to all anchor links with `#` across every page. However, smooth scrolling is only needed on API pages, since the TOC exists only there. Scoping it to `#api-doc` avoids unnecessary event handling and prevents inconsistent anchor `#` behavior on other pages. Also, checking `"#api-doc"` exist on page is required before applying click event.